chat-plugins/datasearch.js

```diff
@@ -1382,7 +1382,8 @@ function runItemsearch(target, cmd, canAll, message) {
 		}
 
 		// iterate over found items again to make sure they all are the best match
-		for (let [i, id] of foundItems.entries()) {
+		// use a reversed copy for iteration so we can remove items mid-loop without changing the remaining indices
+		for (let [i, id] of foundItems.map((element, index) => [index, element]).reverse()) {
```

Comment:
I'd rather not use custom maps when built-ins exist.

In other words: `[...foundItems.entries()].reverse()`

Reply:
Oh I didn't know you could spread do that, nice

Reply:
Yeah, spread is the best way to convert an iterator into an array.